client/src/main.tsx

```diff
@@ -3,39 +3,56 @@ import ReactDOM from "react-dom/client";
 
 import { Login, Register, Loading, NotFound, EditProfile, Search } from "./components/pages";
 
-import { createBrowserRouter, RouterProvider, redirect, Outlet } from "react-router-dom";
+import { createBrowserRouter, RouterProvider, defer } from "react-router-dom";
 
 import { ChakraProvider } from "@chakra-ui/react";
 import { userProfileSchema } from "./schemas.ts";
+import { useProfileStore } from "./store.ts";
+import { getToken } from "./token.ts";
 
-const router = createBrowserRouter([
-  {
-    path: "/",
-    loader: async () => {
-      const token = localStorage.getItem("token") ?? null;
+const implicitLogin = async () => {
+  try {
+    const profile = useProfileStore.getState();
+    if (profile.displayName !== "") return null;
+
+    const token = getToken();
+    if (!token) throw new Error("Token doesn't exist");
 
-      try {
-        if (!token) throw new Error("No such token");
+    const response = await fetch("localhost:8080/implicit_login", {
+      method: "GET",
+      headers: {
+        Authorization: `Bearer ${token}`,
+      },
+    });
 
-        const response = await fetch("localhost:8080/implicit_login", {
-          method: "GET",
-          headers: {
-            Authorization: `Bearer ${token}`,
-          },
-        });
+    const parsedResponse = userProfileSchema.safeParse(response.json());
+    if (!parsedResponse.success) throw new Error("Unable to make implicit login");
```

Comment:
The `safeParse` function uses the schema to parse the data, and if the parsing fails it makes the `success` field false in the returned object.

Here we are checking if `success` is false, and then throwing an error. At that point, we should just use the normal `parse` function, which parses the data and throws an error if it fails. Also, the parse function does not wrap the result in an object containing a `success` field, so it will simplify the code as well.